[crystal-bys] iP

crystal-bys opened the pull request.

crystal-bys committed:
Level 0 completed
- name of bot changed to Crystal
- greets user and exits when run

crystal-bys committed:
Level 0 completed
- name of bot changed to Crystal
- greets user and exits when run

crystal-bys committed:
Level 1 completed. Echo completed.

crystal-bys committed:
Level 2 completed. Add and list completed.

crystal-bys committed:
Level 3 completed. Mark as done and unmark completed.

crystal-bys committed:
A-CodingStandard completed. All indentations are checked and all variables are in camelCase.

crystal-bys committed:
Level 4 completed. Todo, Event and Deadline completed.

crystal-bys committed:
A-CodeQuality completed. Considered KISS and avoiding magic numbers, but realised that was practised earlier already. Removed unused code and considered making the happy path prominent in main but understood that that was unavoidable.

crystal-bys committed:
A-TextUiTesting completed. Edited some printing code and the runtest.bat file as FC was not recognised. Wrote in a few input commands and the expected outputs. Seperated the initial output upon running Crystal.java and the outputs afterwards with a space.

TheDinos commented on src/main/java/Crystal.java:
```diff
@@ -0,0 +1,80 @@
+import java.util.ArrayList;
+
+public class Crystal {
+    public static void main(String[] args) {
```
Perhaps the program flow in "main" could be simplified with the use of helper functions?

TheDinos commented on src/main/java/Crystal.java:
```diff
@@ -0,0 +1,80 @@
+                }
+                System.out.println("____________________________________________________________");
+            } else if (line.startsWith("todo")){
+                String description = line.substring(5).trim();
```
Perhaps the number '5' here could be turned into a constant so that it isn't a "magic number", as with the case for other constant numbers in this code.

TheDinos commented on src/main/java/Crystal.java:
```diff
@@ -0,0 +1,80 @@
+        String line;
+        Scanner in = new Scanner(System.in);
+        ArrayList<Task> tasks = new ArrayList<>();
+        System.out.println("____________________________________________________________\n"
```
Perhaps this string could be turned into a constant to minimise code duplication?

TheDinos commented on src/main/java/Crystal.java:
```diff
@@ -0,0 +1,80 @@
+        line = in.nextLine();
+
+        while (!line.equals("bye")) {
+            if (line.equals("list")) {
```
Perhaps comments could be added for each if-else loop for other programmers to better understand the program flow?

TheDinos reviewed (commented):
Overall, the coding flow is clear and succint.

However, various functions and comments could be added to increase user readability.

wx-03 commented on src/main/java/Crystal.java:
```diff
@@ -0,0 +1,80 @@
+                String by = twoParts[1];
+                Task t = new Deadline(description, by);
+                tasks.add(t);
+                System.out.println("____________________________________________________________\n"
+                        + "OK, I have added this task:\n"
+                        + t + "\n"
+                        + "Now you have " + tasks.size() + " tasks in the list.\n"
+                        + "____________________________________________________________");
```
Good job on line wrapping!

wx-03 commented on src/main/java/Crystal.java:
```diff
@@ -0,0 +1,80 @@
+            line = in.nextLine();
+
+        }
+        System.out.println("____________________________________________________________\n"
+                           + "Adios, hasta luego!\n"
+                           + "____________________________________________________________");
```
Indentation for wrapped lines should be 8 spaces

Gao327 commented on src/main/java/Crystal.java:
```diff
@@ -7,9 +7,9 @@ public static void main(String[] args) {
         Scanner in = new Scanner(System.in);
         ArrayList<Task> tasks = new ArrayList<>();
         System.out.println("____________________________________________________________\n"
```
Same as TheDinos, just extract it out to be a constan.

Gao327 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,14 @@
+public class Deadline extends Task {
```
I like those parts where you use inheritance to make code simpler

crystal-bys committed:
Level 5 completed. Handling of error is done.

crystal-bys committed:
Merge branch 'branch-Level-5'
Merging is for Level-5 increment, handling errors and A-Exceptions have
been completed on the branch-Level-5.

crystal-bys committed:
seperation of files in packages, done.

crystal-bys committed:
Merge branch 'branch-A-Packages'
Merging branch for seperating files into packages. Merge into master
branch.

crystal-bys committed:
Level 6 task completed. Can delete tasks. Fixed some bugs as well.

crystal-bys committed:
Level 7 task completed. Save data to hard disk completed. Fixed some bugs.

crystal-bys committed:
Merge branch 'branch-Level-6'
Merging branch of branch-Level-6 with master as Level-6 is completed.

crystal-bys committed:
Edited the files in Crystal.java to resolve merge conflict.

tzerbin commented on src/main/java/Crystal.java:
```diff
@@ -0,0 +1,298 @@
+import java.util.Scanner;
```
nit: You could arrange java-provided libraries imports together and separate self-defined packages imports.

tzerbin commented on src/main/java/Crystal.java:
```diff
@@ -0,0 +1,298 @@
+    private static final int DEADLINE_CHAR_COUNT = 8;
+    private static final int EVENT_CHAR_COUNT = 5;
+
+    public static void horizontalLine() {
```
Perhaps name your method with a verb instead?
```suggestion
    public static void printHorizontalLine() {
```

tzerbin commented on src/main/java/Crystal.java:
```diff
@@ -0,0 +1,298 @@
+        horizontalLine();
+    }
+
+    public static void mark(int taskNumber) {
```
Could you be more descriptive in this method name (together with the functionality with the unmark method)?

tzerbin commented on src/main/java/Crystal.java:
```diff
@@ -0,0 +1,298 @@
+
+    public static void sayHello() {
+        horizontalLine();
+        System.out.println("Hola! I'm Crystal.\n"
```
Please cut down on the use of magic strings and numbers in the entire codebase!

tzerbin commented on src/main/java/Crystal.java:
```diff
@@ -0,0 +1,298 @@
+        }
+    }
+
+    public static void addEvent(String line) {
```
line can be easily confused with the horizontal line as defined above.

tzerbin commented on src/main/java/Crystal.java:
```diff
@@ -0,0 +1,298 @@
+        }
+    }
+
+    private static void saveTaskList(ArrayList<Task> list) {
```
Perhaps be more descriptive with the parameter name?

tzerbin commented on src/main/java/tasks/Deadline.java:
```diff
@@ -0,0 +1,19 @@
+        this.by = by;
+    }
+
+    public String fileFormat() {
```
Perhaps name your method with a verb instead?

tzerbin commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,35 @@
+        this.isDone = true;
+    }
+
+    public void unmark() {
```
Could you be more descriptive here?

tzerbin commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,35 @@
+        this.isDone = false;
+    }
+
+    public abstract String fileFormat();
```
Perhaps use verbs instead for method names?

tzerbin commented on src/main/java/tasks/Task.java:
```diff
@@ -0,0 +1,35 @@
+
+    public abstract String fileFormat();
+
+    public void updateBool(boolean isDoneUpdated){
```
I believe this is a setter method; perhaps change the name to:
```suggestion
    public void setIsDone(boolean isDoneUpdated){
```

tzerbin reviewed (commented):
Nice attempt in following guidelines!
You could work on naming your variables and methods more descriptively, and reduce the usage of magic numbers / strings.

crystal-bys committed:
debugged for jar file

crystal-bys committed:
renamed some methods to improve code readability

crystal-bys committed:
Rewrote the code to fix some bugs, as well as to make it more OOP

crystal-bys committed:
Level 9 completed. Find keywords is possible.

crystal-bys committed:
JavaDoc done. Minimal level completed.

crystal-bys committed:
Merge pull request #1 from crystal-bys/branch-Level-9

Level 9 completed. Find keywords is possible.

crystal-bys committed:
Javadoc changes.

crystal-bys committed:
javadoc merge conflict done

crystal-bys committed:
javadoc for real done after pulling

crystal-bys committed:
Merge pull request #2 from crystal-bys/branch-A-JavaDoc

Branch a java doc

crystal-bys committed:
User guide updated.

crystal-bys committed:
Added in a help method to display the list of commands, updated the user guide as well

crystal-bys committed:
fixed a mistake in the user guide.

crystal-bys committed:
Merge conflict resolved. For the A-JavaDoc tag

crystal-bys committed:
Fixed some bugs in the new functions, as well as retested the jar file with the new updates.
